dask/dataframe/tests/test_io.py

```diff
@@ -959,6 +959,97 @@ def test_to_hdf_exceptions():
+    with tmpdir() as dn:
+        fn = os.path.join(dn, 'data_*.h5')
+        a.to_hdf(fn, '/data', get=get_sync)
+        out = dd.read_hdf(fn, '/data')
+        eq(df, out)
+
+    # test single file multiple nodes
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data*', get=get_sync)
+        out = dd.read_hdf(fn, '/data*')
+        eq(df, out)
+
+
+def test_to_hdf_thread():
+    pytest.importorskip('tables')
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'],
+                       'y': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]},
+                            index=[1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12., 13., 14., 15., 16.])
+    a = dd.from_pandas(df, 16)
+
+    # test single file single node
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data', get=dask.threaded.get)
+        out = pd.read_hdf(fn, '/data')
+        eq(df, out)
+
+    # test multiple files single node
+    with tmpdir() as dn:
+        fn = os.path.join(dn, 'data_*.h5')
+        a.to_hdf(fn, '/data', get=dask.threaded.get)
+        out = dd.read_hdf(fn, '/data')
+        eq(df, out)
+
+    # test single file multiple nodes
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data*', get=dask.threaded.get)
+        out = dd.read_hdf(fn, '/data*')
+        eq(df, out)
+
+
+def test_to_hdf_process():
+    pytest.importorskip('tables')
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'],
+                       'y': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]},
+                            index=[1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12., 13., 14., 15., 16.])
+    a = dd.from_pandas(df, 16)
+
+    # test single file single node
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data', get=dask.multiprocessing.get, lock=False)
+        out = pd.read_hdf(fn, '/data')
+        eq(df, out)
+
+    # test multiple files single node
+    with tmpdir() as dn:
+        fn = os.path.join(dn, 'data_*.h5')
+        a.to_hdf(fn, '/data', get=dask.multiprocessing.get, lock=False)
+        out = dd.read_hdf(fn, '/data')
+        eq(df, out)
+
+
+@pytest.mark.xfail(reason="can't lock because multiprocess and can't write "
+                          "to single file from multiple processes")
+def test_to_hdf_process_singlefile_multiplenodes():
+    pytest.importorskip('tables')
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'],
+                       'y': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]},
+                            index=[1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12., 13., 14., 15., 16.])
+    a = dd.from_pandas(df, 16)
+
+    # test single file multiple nodes
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data*', get=dask.multiprocessing.get)
+        out = dd.read_hdf(fn, '/data*')
+        eq(df, out)
```

Comment:
You could try a `multiprocessing.Lock` instead or a file based lock such as is found in the `locket` library.